File: TrainTicketPlatformAPI/Models/Payment.cs
using Microsoft.EntityFrameworkCore;

namespace TrainTicketPlatformAPI.Models
{
    public class Payment
    {
        public int Id { get; set; }
        public int? BookingId { get; set; }
        public int? BookingOrderId { get; set; }
        public string PaymentIntentId { get; set; } = string.Empty;
        public string PaymentMethodToken { get; set; } = string.Empty;
        public DateTime PaymentDate { get; set; }
        public string Status { get; set; } = string.Empty;
        [Precision(18, 2)]
        public decimal Amount { get; set; }
        public int LoyaltyPointsRedeemed { get; set; }
        [Precision(18, 2)]
        public decimal LoyaltyDiscountAmount { get; set; }

        public Booking? Booking { get; set; }
        public BookingOrder? BookingOrder { get; set; }
    }
}

File: train-ticket-frontend/src/types/payment.ts
export type PaymentIntent = {
  paymentIntentId: string;
  bookingId: number | null;
  bookingOrderId: number | null;
  bookingIds: number[];
  originalAmount: number;
  amount: number;
  loyaltyPointsRedeemed: number;
  loyaltyDiscountAmount: number;
  currency: string;
  status: string;
  expiresAtUtc: string | null;
  testPaymentMethodTokens: string[];
};

export type Payment = {
  id: number;
  bookingId: number | null;
  bookingOrderId: number | null;
  bookingIds: number[];
  paymentIntentId: string;
  paymentDate: string;
  status: string;
  amount: number;
  loyaltyPointsRedeemed: number;
  loyaltyDiscountAmount: number;
};
